statsd/src/main/java/com/avast/metrics/statsd/StatsDMetricsMonitorBuilder.java

```diff
@@ -0,0 +1,79 @@
+package com.avast.metrics.statsd;
+
+import com.avast.metrics.api.Naming;
+import com.avast.metrics.filter.MetricsFilter;
+
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class StatsDMetricsMonitorBuilder {
+		private String host;
+		private int port;
+		private boolean autoRegisterMetrics = false;
+		private String prefix;
+		private Naming naming = Naming.defaultNaming();
+		private Duration gaugeSendPeriod = getDefaultGaugeSendPeriod();
+		private ScheduledExecutorService scheduler = createScheduler();
+		private MetricsFilter metricsFilter = MetricsFilter.ALL_ENABLED;
+
+		public StatsDMetricsMonitorBuilder setHost(String host) {
+				this.host = host;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setPort(int port) {
+				this.port = port;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setAutoRegisterMetrics(boolean autoRegisterMetrics) {
+				this.autoRegisterMetrics = autoRegisterMetrics;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setPrefix(String prefix) {
+				this.prefix = prefix;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setNaming(Naming naming) {
+				this.naming = naming;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setGaugeSendPeriod(Duration gaugeSendPeriod) {
+				this.gaugeSendPeriod = gaugeSendPeriod;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setScheduler(ScheduledExecutorService scheduler) {
+				this.scheduler = scheduler;
```

Comment:
Resource leak of existing scheduler and its threads, there is already the default one. What is the designed ownership? Who should release the scheduler? It isn't obvious from the code, it's mixed.